src/_pytest/python_api.py

```diff
@@ -536,18 +543,40 @@ def approx(expected, rel=None, abs=None, nan_ok: bool = False) -> ApproxBase:
 
 
 def _is_numpy_array(obj: object) -> bool:
```

Comment:
To avoid duplicating the logic, I believe `_is_numpy_array` can be implemented in terms of `_as_numpy_array`:

```
def _is_numpy_array(obj: object) -> bool:
    """
    Return true if the given object is implicitly convertible to numpy array, 
    and numpy is already imported .
    """
    return _as_numpy_array(obj) is not None
```

Note I've also changed the docstring: the method won't really import `numpy` at all, so it is better to rectify that.

Reply:
Good points both; now fixed.